internal/equality/equality_test.go

```diff
@@ -550,12 +550,39 @@ func TestLoadBalancerServiceChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "if load balancer IP changed",
+			mutate: func(svc *corev1.Service) {
+				svc.Spec.LoadBalancerIP = "5.6.7.8"
+			},
+			expect: true,
+		},
 	}
 
 	for _, tc := range testCases {
+
 		cntr.Spec.NetworkPublishing.Envoy.Type = operatorv1alpha1.LoadBalancerServicePublishingType
 		cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.Scope = operatorv1alpha1.ExternalLoadBalancer
 		cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type = operatorv1alpha1.AWSLoadBalancerProvider
+		if tc.description == "if load balancer IP changed" {
```

Comment:
might be ok to leave now, but would be good to refactor this test suite a bit

we're currently taking a global contour variable and modifying it in each test case, then generating a service from that contour, would be good to not use that global variable if we are going to modify it and for this test just create a service object to check the equality against with the copy/mutate pattern which is there and nice to avoid duplication

Reply:
I was thinking the same. It would be quite a change, as it would affect whole `equality_testo.go`, so  I thought we can set this up separately.